google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingImpl.java

```diff
@@ -797,6 +799,76 @@ public void write(Iterable<LogEntry> logEntries, WriteOption... options) {
     inWriteCall.set(true);
 
     try {
+      final Map<Option.OptionType, ?> writeOptions = optionMap(options);
+      final Boolean populateMetadata1 = getOptions().getAutoPopulateMetadata();
+      final Boolean populateMetadata2 =
+          WriteOption.OptionType.AUTO_POPULATE_METADATA.get(writeOptions);
+      if (populateMetadata2 == Boolean.TRUE
+          || (populateMetadata2 == null && populateMetadata1 == Boolean.TRUE)) {
+        final Boolean needDebugInfo =
+            Iterables.any(
+                logEntries,
+                log -> log.getSeverity() == Severity.DEBUG && log.getSourceLocation() == null);
+        final SourceLocation sourceLocation =
+            needDebugInfo ? SourceLocation.fromCurrentContext(1) : null;
+        final MonitoredResource sharedResourceMetadata = RESOURCE.get(writeOptions);
+        final MonitoredResource resourceMetadata =
+            sharedResourceMetadata == null ? MonitoredResourceUtil.getResource(null, null) : null;
+        final Context context = (new ContextHandler()).getCurrentContext();
+        final ArrayList<LogEntry> populatedLogEntries = Lists.newArrayList();
+        for (LogEntry entry : logEntries) {
+          LogEntry.Builder builder = entry.toBuilder();
+          if (resourceMetadata != null && entry.getResource() == null) {
+            builder.setResource(resourceMetadata);
+          }
+          if (context != null && entry.getHttpRequest() == null) {
+            builder.setHttpRequest(context.getHttpRequest());
```

Comment:
Should you also check if `context.getHttpRequest() != null`?

Reply:
Adding such check does not change the flow. Neither it improves a performance nor security of the execution. Similar logic is applied for if`ed setups of other metadata.